### PR TITLE
Fix DBFormatter unit test

### LINKED ISSUE #10231: [py2py3] Potential bug in /Database/DBFormatter.py
**How to reproduce it**

We have not seen any bug yet, but it is possible that the changes to `Database/DBFormatter.py` introduced in #10057 have some breaking side effect.

**Describe the bug**

The potential bug is described [in these comments](https://github.com/dmwm/WMCore/pull/10057#discussion_r539285754).

Overview: `DBFormatter.formatDict()`

```python
+ from builtins import str, bytes, zip, range
[...]
- if isinstance(i[index], unicode): 
- entry[str(descriptions[index].lower())] = str(i[index]) 
+ if isinstance(i[index], str):
+ entry[str(descriptions[index].lower())] = bytes(i[index])
[...]
```

These changes have not been tested, since `test/python/WMCore_t/Database_t/DBFormatter_t.testBFormatting()` is marked as integration (when I try to run it in my docker it hangs).

**Impact of the bug**

Grepping `formatDict()` returns many occurrences. If this changes cause problems, we should not wait long before finding them!

**Expected behavior**

This modernization should make `DBFormatter` compatible with both py2 and py3. Moreover, it should behave in both py2 and py3 as it behaved in py2 before #10057

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #10231 

#### Status
ready

#### Description
Update the DBFormatter unit test to use the already defined test database/tables and fixed it to work with both MySQL and Oracle.

I have also executed all the unit tests in (they execute tons of SQL queries!):
```
nosetests test/python/WMCore_t/WMBS_t/Subscription_t.py
nosetests test/python/WMCore_t/WMBS_t/Workflow_t.py
```

and they all work fine in both Oracle and MySQL in python3. So, I'd say the DBFormatter src changes did not introduce any defects.

UPDATE: second commit contains a couple of unstable python3 unit tests.

#### Is it backward compatible (if not, which system it affects?)
yes

#### Related PRs
none

#### External dependencies / deployment changes
none

### CODE REVIEW

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
Marking more py3 tests as unstable

File: test/etc/UnstableTests.txt [hunk 1]
```diff
@@ -38,4 +38,8 @@ WMCore_t.WorkQueue_t.DataStructs_t.WorkQueueElement_t.WorkQueueElementTest:testP
 WMCore_t.WMBS_t.JobSplitting_t.FileBased_t.FileBasedTest:testLocationSplit
 WMCore_t.Services_t.LogDB_t.LogDB_t.LogDBTest:test_heartbeat
 WMCore_t.WorkQueue_t.DataStructs_t.WorkQueueElement_t.WorkQueueElementTest:testPossibleSitesLocationFlags
-WMCore_t.WorkQueue_t.DataStructs_t.WorkQueueElement_t.WorkQueueElementTest:testPossibleSites
\ No newline at end of file
+WMCore_t.WorkQueue_t.DataStructs_t.WorkQueueElement_t.WorkQueueElementTest:testPossibleSites
+WMCore_t.Configuration_t.ConfigurationTest:testF
```
Comment (mapellidario) [inline test/etc/UnstableTests.txt:42]:
```suggestion
-WMCore_t.Configuration_t.ConfigurationTest:testF
```

I would remove this line, I saw and fixed this while working on #10575. For the time being, the fix is in #10575, but we can move it to a separate PR

**Review by mapellidario — APPROVED** [review]
Thanks Alan! These changes look perfect! 

I made only a single comment in the list of unstable unittests, but you can ignore it and postpone that to a later stage, it's not critical. Thanks again!

File: test/etc/UnstableTests.txt (hunk 1 above)
Comment (amaltaro) [reply test/etc/UnstableTests.txt:42]: I can take it out from this PR then. New change coming in in a minute.

Comment (mapellidario) [reply test/etc/UnstableTests.txt]: perfect, thanks!

**Comment by amaltaro** [comment]
Thanks Dario.
